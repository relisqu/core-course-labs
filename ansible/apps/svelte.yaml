---
- name: Deploy svelte time app
  hosts: vk_cloud
  become: true
  roles:
   - role: roles/web_app
     image: relisqu/svelte-app:latest
     ports:
      - "5173:5173"