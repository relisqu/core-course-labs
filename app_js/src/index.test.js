import { describe, it, expect, test } from 'vitest';
import { cleanup, fireEvent, render, screen } from '@testing-library/svelte';
import App from './routes/+page.svelte';

import axios from 'axios';
import moment from 'moment-timezone';

describe('Hello.svelte', () => {
	// TODO: @testing-library/svelte claims to add this automatically but it doesn't work without explicit afterEach

	it('renders', () => {
		const { container } = render(App);
		expect(container).toBeTruthy();
	});

	it('mounts', async () => {
		const { container } = render(App);
		expect(container).toBeTruthy();
		const response = await axios.get('http://worldtimeapi.org/api/timezone/Europe/Moscow');
		const currentMoscowTime = moment(response.data.datetime).format('HH:mm');
		expect(container.innerHTML).toContain(moment().tz('Europe/Moscow').format('HH:mm:ss'));
	});
});
