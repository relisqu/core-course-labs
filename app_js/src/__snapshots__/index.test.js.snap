// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hello.svelte > mounts 1`] = `undefined`;
